glacier/src/exporters/state.ts

```diff
@@ -0,0 +1,24 @@
+import redux = require("redux");
+import JSZip = require("jszip");
+import {ModelState} from "../";
+import {Exporter, createSvgExporter} from "./";
+
+export function createStateExporter(store: redux.Store<ModelState>, uuid: number, libary: Buffer) {
+    const updater: Exporter<any> = ((() => {
+        // On update...
+        // store.getState()
+    }) as Exporter<any>);
+    updater.export = () => {
+        return new Promise<JSZip>((resolve, reject) => {
+            let stateString = JSON.stringify(store.getState());
+            const svgExporter = createSvgExporter(store, uuid);
+            let zip = new JSZip();
+            zip.file("state.json", stateString);
+            zip.file("thumnail.svg", svgExporter.export());
```

Comment:
I get the following error when I implement the change.
`error TS1308: 'await' expression is only allowed within an async function.`